### PR TITLE
Various radiko enhancements

### LINKED ISSUE #6090: Broken site: radiko.jp: Invalid data found when processing input
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2023.01.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Japan

### Provide a description that is worded well enough to be understood

Downloading the radiko.jp is broken.
Error message: `Invalid data found when processing input`
It's not specific program but any in radiko.jp.

Sample program: https://radiko.jp/#!/ts/IBS/20230125050000 (Available only for one week)

Following is the example of medialist (checked by Chrome developer tools):

```
#EXTM3U
#EXT-X-VERSION:6
#EXT-X-TARGETDURATION:5
#EXT-X-ALLOW-CACHE:NO
#EXT-X-MEDIA-SEQUENCE:4230442
#EXT-X-DISCONTINUITY-SEQUENCE:0
#EXT-X-START:TIME-OFFSET=0
#EXT-X-PROGRAM-DATE-TIME:2023-01-25T05:00:00.002+09:00
#EXTINF:5.035,
https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/segments/o/B/IBS/20230125/20230125_050000_ycnag.aac
#EXT-X-PROGRAM-DATE-TIME:2023-01-25T05:00:05.037+09:00
#EXTINF:5.035,
https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/segments/o/B/IBS/20230125/20230125_050005_t56hm.aac
#EXT-X-PROGRAM-DATE-TIME:2023-01-25T05:00:10.072+09:00
#EXTINF:5.035,
https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/segments/o/B/IBS/20230125/20230125_050010_l8ufg.aac
```

Each flagment aac is no problem to download and play:

```console
$ yt-dlp -vU https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/segments/o/B/IBS/20230125/20230125_050000_ycnag.aac
[debug] Command-line config: ['-vU', 'https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/segments/o/B/IBS/20230125/20230125_050000_ycnag.aac']
[debug] Encodings: locale cp932, fs utf-8, pref cp932, out utf-8, error utf-8, screen utf-8 
[debug] yt-dlp version 2023.01.06 [6becd25] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021) 
[debug] exe versions: ffmpeg 5.1.2-essentials_build-www.gyan.dev (setts), ffprobe 5.1.2-essentials_build-www.gyan.dev 
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4 
[debug] Proxy map: {} 
[debug] Loaded 1760 extractors 
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest 
Latest version: 2023.01.06, Current version: 2023.01.06 
yt-dlp is up to date (2023.01.06) 
[generic] Extracting URL: https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/segments/o/B/IBS/20230125/20230125_050000_ycnag.aac 
[generic] 20230125_050000_ycnag: Downloading webpage 
WARNING: [generic] Falling back on generic information extractor 
WARNING: [generic] URL could be a direct video link, returning it as such. 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 20230125_050000_ycnag: Downloading 1 format(s): 0
[debug] Invoking http downloader on "https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/segments/o/B/IBS/20230125/20230125_050000_ycnag.aac"
[debug] File locking is not supported. Proceeding without locking 
[download] Destination: 20230125_050000_ycnag [20230125_050000_ycnag].aac 
[download] 100% of   30.53KiB in 00:00:00 at 337.67KiB/s
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU 'https://radiko.jp/#!/ts/IBS/20230125050000'
[debug] Command-line config: ['-vU', 'https://radiko.jp/#!/ts/IBS/20230125050000']
[debug] Encodings: locale cp932, fs utf-8, pref cp932, out utf-8, error utf-8, screen utf-8 
[debug] yt-dlp version 2023.01.06 [6becd25] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021) 
[debug] exe versions: ffmpeg 5.1.2-essentials_build-www.gyan.dev (setts), ffprobe 5.1.2-essentials_build-www.gyan.dev 
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4 
[debug] Proxy map: {}
[debug] Loaded 1760 extractors 
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest 
Latest version: 2023.01.06, Current version: 2023.01.06 
yt-dlp is up to date (2023.01.06) 
[Radiko] Extracting URL: https://radiko.jp/#!/ts/IBS/20230125050000 
[Radiko] 20230125050000: Downloading radio program for IBS station 
[debug] Loading radiko.auth_data from cache 
[Radiko] 20230125050000: Downloading stream information 
[Radiko] 20230125050000: Downloading m3u8 information from si-c-radiko.smartstream.ne.jp 
[Radiko] 20230125050000: Downloading m3u8 information from si-f-radiko.smartstream.ne.jp 
[Radiko] 20230125050000: Downloading m3u8 information from tf-f-rpaa-radiko.smartstream.ne.jp 
[Radiko] 20230125050000: Downloading m3u8 information from tf-c-rpaa-radiko.smartstream.ne.jp 
[Radiko] 20230125050000: Downloading m3u8 information from rd-wowza-radiko.radiko-cf.com 
[Radiko] 20230125050000: Downloading m3u8 information from rd-wowza-radiko.radiko-cf.com 
[Radiko] 20230125050000: Downloading m3u8 information from radiko.jp 
[Radiko] 20230125050000: Downloading m3u8 information from f-radiko.smartstream.ne.jp 
[Radiko] 20230125050000: Downloading m3u8 information from c-radiko.smartstream.ne.jp 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id 
[debug] Default format spec: best/bestvideo+bestaudio 
[info] 20230125050000: Downloading 1 format(s): tf-f-rpaa-radiko.smartstream.ne.jp
[debug] Invoking ffmpeg downloader on "https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/medialist?session=1.3B3ssFKL3TRfcEaqBrGEU4&station_id=IBS" 
[download] Destination: テスト音楽 2023-01-26 00_15 [20230125050000].m4a
[debug] ffmpeg command line: ffmpeg -y -loglevel verbose -headers "User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.74 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-us,en;q=0.5
Sec-Fetch-Mode: navigate
" -i "https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/medialist?session=1.3B3ssFKL3TRfcEaqBrGEU4&station_id=IBS" -c copy -f mpegts "file:テスト音楽 2023-01-26 00_15 [20230125050000].m4a.part"
ffmpeg version 5.1.2-essentials_build-www.gyan.dev Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 12.1.0 (Rev2, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
[tcp @ 000002b18eeb1dc0] Starting connection attempt to 203.211.199.163 port 443
[tcp @ 000002b18eeb1dc0] Successfully connected to 203.211.199.163 port 443
[AVIOContext @ 000002b18f294cc0] Statistics: 6 bytes read, 0 seeks
https://tf-f-rpaa-radiko.smartstream.ne.jp/tf/medialist?session=1.3B3ssFKL3TRfcEaqBrGEU4&station_id=IBS: Invalid data found when processing input
 

ERROR: ffmpeg exited with code 1 
  File "yt_dlp\__main__.py", line 17, in <module>
  File "yt_dlp\__init__.py", line 970, in main
  File "yt_dlp\__init__.py", line 960, in _real_main
  File "yt_dlp\YoutubeDL.py", line 3358, in download
  File "yt_dlp\YoutubeDL.py", line 3333, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1491, in extract_info
  File "yt_dlp\YoutubeDL.py", line 1502, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1599, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1658, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 2772, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 3236, in process_info
  File "yt_dlp\YoutubeDL.py", line 2959, in dl
  File "yt_dlp\downloader\common.py", line 444, in download
  File "yt_dlp\downloader\external.py", line 76, in real_download
  File "yt_dlp\YoutubeDL.py", line 1012, in report_error
  File "yt_dlp\YoutubeDL.py", line 941, in trouble
```

### PR DESCRIPTION
**Pull request by garret1317**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

These patches improve the `radiko` extractor using things I've learnt while developing my plugin [yt-dlp-rajiko](https://github.com/garret1317/yt-dlp-rajiko).
Hopefully the commit messages explain each change well enough, feel free to ask if not.

e9117f0bab990ad43c1a94b3c388bc7d10437b6b sort-of-fixes #6090, since the new format downloads with yt-dlp's hls downloader rather than ffmpeg.
However it could possibly break someone's workflow, as the extractor now extracts completely different formats. This commit can be skipped if you'd like.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🌐🕒🔑

Enhance the `radiko.py` extractor to handle different types of streams, check geo-restrictions, and streamline the code.

> _`Radiko` is the sound of doom, the voice of the oppressed_
> _We break the chains of geo-restriction, we stream the timefree past_
> _We simplify the code of darkness, we decrypt the hidden light_
> _We fix the flaws of format extraction, we make the metadata right_

### Walkthrough
*  Disable geo-bypass and raise error for out-of-Japan users ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fR17), [link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL61-R79))
*  Rename and move `_auth_client` method to `_negotiate_token` ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL35-R37))
*  Use different XML endpoint and filter URLs by `timefree` attribute for live and timefree streams ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL104-R139), [link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL141-R157))
*  Generate random `lsid` parameter for playlist URL ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fR2), [link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL104-R139))
*  Remove `live` argument from `_extract_m3u8_formats` method call ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL130-R146))
*  Simplify token and area ID extraction and return formats early ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL169-R194), [link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL180-R204))
*  Add comment to clarify hardcoded full key fallback ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL78-R93))
*  Change `is_live` key to `live_status` and set it to `was_live` for timefree streams ([link](https://github.com/yt-dlp/yt-dlp/pull/8221/files?diff=unified&w=0#diff-0b9132bf887faf956d81399a6a1e08e33718c64eb4021379f3d3468f76b2355fL180-R204))

</details>

### CODE REVIEW

**Comment by garret1317** [comment]
actually yes do skip e9117f0bab990ad43c1a94b3c388bc7d10437b6b
pc_html5 api does actually return the not-live streams, its just a matter of prioritising them over the live ones

File: yt_dlp/extractor/radiko.py [hunk 1]
```diff
@@ -101,33 +116,34 @@ def _find_program(self, video_id, station, cursor):
 
     def _extract_formats(self, video_id, station, is_onair, ft, cursor, auth_token, area_id, query):
         m3u8_playlist_data = self._download_xml(
-            f'https://radiko.jp/v3/station/stream/pc_html5/{station}.xml', video_id,
+            f'https://radiko.jp/v3/station/stream/aSmartPhone7a/{station}.xml', video_id,
```
Comment (bashonly) [inline yt_dlp/extractor/radiko.py:119]: > actually yes do skip https://github.com/yt-dlp/yt-dlp/commit/e9117f0bab990ad43c1a94b3c388bc7d10437b6b
> pc_html5 api does actually return the not-live streams, its just a matter of prioritising them over the live ones

Do you mean we want to revert this?

File: yt_dlp/extractor/radiko.py [hunk 2]
```diff
@@ -166,31 +182,26 @@ def _real_extract(self, url):
...
-                    'ft': radio_begin,
-                    'end_at': radio_end,
-                    'to': radio_end,
-                    'seek': video_id,
-                })
-            if formats:
-                break
-
+        auth_token, area_id = self._auth_client()
         return {
```
Comment (bashonly) [inline yt_dlp/extractor/radiko.py:186]:
```suggestion
        auth_token, area_id = self._auth_client()

        return {
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/radiko.py (hunk 1 above)
Comment (garret1317) [reply yt_dlp/extractor/radiko.py:119]: yes
i'll revert that commit and then apply your other fixes
~~cant be bothered to~~ not sure how id go about prioritising the not-live so i'll just leave it for now

**Comment by garret1317** [comment]
oh bugger ive broken the indentation

**Review by bashonly — APPROVED** [review]

**Review by Grub4K — APPROVED** [review]

**Pull request merged**